src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -274,6 +277,28 @@ private static boolean inLambda(DetailAST paramDef) {
     }
 
     /**
+     * if definition and initialization are in different for-loop, it can not be declared
+     * final. In that case, it will return false.
+     * @param ast1 VARIABLE_DEF
+     * @param ast2 init variable
+     * @return true if it is initiaized only once
+     */
+    private boolean defAndInitInDifferentLoop(DetailAST ast1, DetailAST ast2) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method defAndInitInDifferentLoop() has an NPath complexity of 48](https://www.codacy.com/app/checkstyle/checkstyle/file/1022207894/issues/source?bid=2193633&fileBranchId=2872415#l286)